cmd/kaspactl/commands.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/kaspanet/kaspad/infrastructure/network/netadapter/server/grpcserver/protowire"
+)
+
+var commandTypes = []reflect.Type{
+	reflect.TypeOf(protowire.KaspadMessage_AddPeerRequest{}),
+	reflect.TypeOf(protowire.KaspadMessage_GetConnectedPeerInfoRequest{}),
+	reflect.TypeOf(protowire.KaspadMessage_GetPeerAddressesRequest{}),
+	reflect.TypeOf(protowire.KaspadMessage_GetCurrentNetworkRequest{}),
+
+	reflect.TypeOf(protowire.KaspadMessage_GetBlockRequest{}),
+	reflect.TypeOf(protowire.KaspadMessage_GetBlocksRequest{}),
+	reflect.TypeOf(protowire.KaspadMessage_GetHeadersRequest{}),
+	reflect.TypeOf(protowire.KaspadMessage_GetBlockCountRequest{}),
+	reflect.TypeOf(protowire.KaspadMessage_GetBlockDagInfoRequest{}),
+	reflect.TypeOf(protowire.KaspadMessage_GetSelectedTipHashRequest{}),
+	reflect.TypeOf(protowire.KaspadMessage_GetVirtualSelectedParentBlueScoreRequest{}),
+	reflect.TypeOf(protowire.KaspadMessage_GetVirtualSelectedParentChainFromBlockRequest{}),
+	reflect.TypeOf(protowire.KaspadMessage_ResolveFinalityConflictRequest{}),
+
+	reflect.TypeOf(protowire.KaspadMessage_GetBlockTemplateRequest{}),
+	reflect.TypeOf(protowire.KaspadMessage_SubmitBlockRequest{}),
+
+	reflect.TypeOf(protowire.KaspadMessage_GetMempoolEntryRequest{}),
+	reflect.TypeOf(protowire.KaspadMessage_GetMempoolEntriesRequest{}),
+	reflect.TypeOf(protowire.KaspadMessage_SubmitTransactionRequest{}),
+
+	reflect.TypeOf(protowire.KaspadMessage_GetUtxosByAddressesRequest{}),
+}
+
+type commandDescription struct {
+	name       string
+	parameters []*parameterDescription
+	typeof     reflect.Type
+}
+
+type parameterDescription struct {
```

Comment:
Why not just use `StructField`?

Reply:
I'd rather not, StructField comes with a lot of strings attached and a lot of redundant stuff.